### PR TITLE
build(odsp-end-to-end-tests): convert to esm

### PR DESCRIPTION
**Pull request by DLehenbauer**
Mechanical conversion to esm

### CODE REVIEW

File: packages/service-clients/end-to-end-tests/odsp-client/.mocharc.cjs [hunk 1]
```diff
@@ -5,7 +5,8 @@
 
 "use strict";
 
-const packageDir = `${__dirname}/../..`;
-const getFluidTestMochaConfig = require("@fluid-internal/mocha-test-setup/mocharc-common.js");
+const getFluidTestMochaConfig = require("@fluid-internal/mocha-test-setup/mocharc-common");
+
+const packageDir = __dirname;
 const config = getFluidTestMochaConfig(packageDir);
 module.exports = config;
```
Comment (DLehenbauer) [inline packages/service-clients/end-to-end-tests/odsp-client/.mocharc.cjs:12]: Moved .mocharc.cjs to the package root.

File: packages/service-clients/end-to-end-tests/odsp-client/package.json [hunk 2]
```diff
@@ -78,6 +78,7 @@
 		"uuid": "^9.0.0"
 	},
 	"devDependencies": {
+		"@arethetypeswrong/cli": "^0.13.3",
```
Comment (jason-ha) [inline packages/service-clients/end-to-end-tests/odsp-client/package.json:81]: unused

File: packages/service-clients/end-to-end-tests/odsp-client/package.json [hunk 3]
```diff
@@ -86,6 +87,7 @@
 		"@types/node": "^18.19.0",
 		"@types/uuid": "^9.0.2",
 		"c8": "^8.0.1",
+		"copyfiles": "^2.4.1",
```
Comment (jason-ha) [inline packages/service-clients/end-to-end-tests/odsp-client/package.json:90]: unused

**Review by jason-ha — CHANGES_REQUESTED** [review]

**Review by jason-ha — CHANGES_REQUESTED** [review]
lockfile needs updated

**Review by jason-ha — APPROVED** [review]

**Pull request merged**